samples/net/wifi/boards/frdm_rw612.conf

```diff
@@ -84,7 +88,7 @@ CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT=y
 CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE=y
 CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP=y
 CONFIG_WIFI_NM_WPA_SUPPLICANT_EAPOL=y
-CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_MBEDTLS_PSA=y
+CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_MBEDTLS_PSA=n
```

Comment:
Do we need to manually set this to `n`, or could we just remove this line? It seems like this will be disabled by default.

Reply:
I was going to ask for clarity on this from anyone who is more familiar with it. The other changes I've made in this PR have a logical basis + can be shown from the DT files, but this one in particular was more of a brute force approach. I'm not sure if mbedtls/entropy initialization is just being called incorrect/hasn't been updated to support this convention (like maybe it has been but is on a feature branch and not mainstream zephyr), and maybe should be added again later for security reasons. I made this PR simply to fix basic functionality of this board for this sample (and we were still concerned the drivers might not have been added correctly from NXP yet but that appears to not be the case).

Removing it could make sense to me, but in the grand scheme of things I was assuming that changes like that should be another PR. That sample/conf file has a few other Kconfig flags being set that I assume are to support future features, as some of them have extremely limited documentation or are not used in Zephyr code as far as I could see. So my thinking was to leave this as `n` for now so that someone might know/remember to ensure that feature does work later - if it was intended correctly by someone or will be supported in the future. Hope that makes sense.

Reply:
Entropy should be using the PSA APIs not the legacy MbedTLS APIs.

Reply:
Ok, I think that makes sense, I'm not fully familiar with what the difference there is (on the implementation details side of things). Do you mind expanding on that more?

@krish2718 Or do you have any input on whether that Kconfig flag should persist here, and/or is it a library change that needs to happen unrelated to this sample? I'm just wondering how that difference affects the scope/content of this PR. Thanks.

Reply:
I don't know the original issues, so, cannot comment much. But we shouldn't disable PSA, instead we should fix the entropy to use PSA. Can you please share details or logs of the original problem?

Reply:
Yeah, I just finished writing an issue for this PR, I can link it to you. I'll copy in the relevant logs here in case it helps at all. 

The bootup logs are as below:
```
Set BGA tx power table data 
[00:00:01.220,419] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN: received event 14
uart:~$ *** Booting Zephyr OS build v4.0.0-rc1-76-g743761d7d1d7 ***
[00:00:01.237,361] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN initialized
uart:~$ supp_psa_crypto_init failed [00:00:01.251,870] <inf> wifi_supplicant: wpa_supplicant initialized
[00:00:01.265,358] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN: received event 16
[00:00:01.276,122] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN: PS_ENTER
[00:00:01.286,156] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN: received event 16
[00:00:01.296,955] <dbg> nxp_wifi: nxp_wifi_wlan_event_callback: WLAN: PS_ENTER
[00:00:01.308,682] <err> wpa_supp: Init of random number generator failed
```
Which corresponds to code in `~/zephyrproject/modules/hostap/src/supp_main.c:1126`:
```
#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_MBEDTLS_PSA
	supp_psa_crypto_init();
#endif
```
which calls `psa_crypto_init()` in `~/zephyrproject/modules/lib/hostap/port/mbedtls/supp_psa_api.c:468`. 

Hope this helps. My initial impression is that a real fix for this aspect is out of scope for this PR, but I'm happy to help or try to fix it you think it is worth doing here.

Reply:
Thanks. `mbedtls_ctr_drbg_seed` CTR-DRBG is a legacy way to get random numbers, so, when PSA is enabled it should using PSA APIs. Anyways, as you say fixing it properly is out of scope of this PR. FYI @MaochenWang1  @fengming-ye from a quick look we need to add proper guards to compiled out legacy stuff  when PSA is enabled. This could affect multiple platforms, so, please take a look
@andrewnyland I would suggest to raise another bug for getting Wi-Fi connection working with PSA enabled.

Reply:
@krish2718 Ok, good to know - thanks for the input. I'll make some notes and make one next week, I have another PR I'd like to submit for the wifi shell that I think I'll do first. And partially as I'm more unfamiliar with that realm (the base security system on-board with mbedtls, I've done a lot with TCP conns recently however). I'll try to at least make an issue for it this week.

Would you suggest removing that line for this PR or just deal with it later?

Reply:
>Would you suggest removing that line for this PR or just deal with it later?

Does it work without that line? If no, then we can leave that in.